src/components/templates/Jazzbar/JazzTab/JazzTab.tsx

```diff
@@ -89,15 +72,23 @@ const Jazz: React.FC<JazzProps> = ({ setUserList, venue }) => {
   const dispatch = useDispatch();
   const venueId = useVenueId();
 
-  const reactionClicked = (user: UserInfo, reaction: EmojiReactionType) => {
-    dispatch(
-      addReaction({
-        venueId,
-        reaction: createReaction({ reaction }, user),
-      })
-    );
-    setTimeout(() => (document.activeElement as HTMLElement).blur(), 1000);
-  };
+  // @debt de-duplicate this with version in src/components/templates/Audience/Audience.tsx
+  const reactionClicked = useCallback(
+    (emojiReaction: EmojiReactionType) => {
+      if (!venueId || !userWithId) return;
+
+      dispatch(
+        addReaction({
+          venueId,
+          reaction: createEmojiReaction(emojiReaction, userWithId),
+        })
+      );
+
+      // @debt Why do we have this here..? We probably shouldn't have it/need it? It's not a very Reacty thing to do..
+      setTimeout(() => (document.activeElement as HTMLElement).blur(), 1000);
```

Comment:
😲😲😲😲. It's not healthy to review things just before going to bed.

[offtop], feel free to resolve 😄